Utils/github_workflow_scripts/autobump_rn.py

```diff
@@ -0,0 +1,1170 @@
+        1. Bumps numeric version of the pack
+        2. Writes new version to metadata
+        3. Creates new release notes file
+        4. Writes previous release notes content to new path
+        5. If there breaking changes file, updating it to the new version
+        Returns: (str) new pack version.
+        """
+
+        print(f"Starting to bump packs {self.pack_id} version.")
+        print(f"Update type: {self._update_type}, Previous RN path: {self._last_rn_file_path}, Is BC: {self._has_bc}.")
+        new_version, metadata_dict = self._update_rn_obj.bump_version_number()
+        self._update_rn_obj.write_metadata_to_file(metadata_dict=metadata_dict)
+        new_release_notes_path = self._update_rn_obj.get_release_notes_path(new_version)
+
+        if Path(new_release_notes_path).stem != self._last_rn_file_path.stem:
+            with open(new_release_notes_path, "w") as fp:
+                fp.write(self._rn_text)
+                if Path(new_release_notes_path).read_text() == self._rn_text:
+                    os.remove(self._last_rn_file_path)
+
+            if self._has_bc:
+                with open(new_release_notes_path.replace("md", "json"), "w") as fp:
+                    fp.write(self._bc_text)
+
+                if self._bc_file.read_text() == self._bc_text:
+                    # delete previous bc file, if it was not changed after merge from master
+                    os.remove(self._bc_file)
+        return new_version
+
+
+class BranchAutoBumper:
+    def __init__(
+        self,
+        pr: PullRequest,
+        git_repo: Repo,
+        packs_to_autobump: List[PackAutoBumper],
+        run_id: str,
+    ):
+        """
+        Args:
+            pr: Pull Request related to the branch.
+            git_repo: Git API object
+            packs_to_autobump: Pack that was changed in this PR and need to autobump its versions.
+            run_id: GitHub action run id.
+        """
+        assert (
+            packs_to_autobump
+        ), f"packs_to_autobump in the pr: {pr.number}, cant be empty."
+        self.pr = pr
+        self.branch = pr.head.ref
+        self.git_repo = git_repo
+        self.packs_to_autobump = packs_to_autobump
+        self.github_run_id = run_id
+
+    def autobump(self):
+        """AutoBumps version for all relevant packs in the pr:
+        1. Checkouts the branch and saves pr changed related data.
+        2. Merges from BASE and accept `theirs` changes.
+        3. AutoBumps version for each relevant packs.
+        4. Commit changes for each pack.
+        5. Comment on the PR.
+        6. Pushes the changes.
+        """
+        body = PR_COMMENT_TITLE.format(self.github_run_id)
+        if self.branch not in ["conflict_in_cs", "conflicts_in_base", "conflict_in_xdr",
+                               "conflict_in_nx", "conflicts_in_msg", "conflicts-in-snow", "conflict_in_base_bc",
+                               "conflicts-inaws"]:
+            # todo: delete it, only for testing
```

Comment:
.

Reply:
I will delete it after pr has been done. I am still running tests on it. All these are my branches to check that the auto bump works :)